src/CLaSH/Signal/Internal.hs

```diff
@@ -269,7 +269,7 @@ register# _ i s = i :- s
 regEn# :: SClock clk -> a -> Signal' clk Bool -> Signal' clk a -> Signal' clk a
 regEn# _ = go
   where
-    go o (e :- es) as@(~(x :- xs)) =
+    go o ~(e :- es) as@(~(x :- xs)) =
```

Comment:
Hmmm.  I had convinced myself that this was not necessary, because the first thing the RHS of the outer `:-` does when reduced is test `e` with an `if` -- implying that both `e :- es` and `e` itself had been reduced to WHNF.  However, we are in subtle enough territory here that I defer to your judgment.

I feel like the pattern used in this function is getting awkward, I may also take a crack at simplifying it.